liblumen_alloc/src/erts/term/map.rs

```diff
@@ -43,6 +43,15 @@ impl Map {
         self.value.get(&key).copied()
     }
 
+    pub fn take(&self, key: Term) -> Option<(Term, HashMap<Term, Term>)> {
+        let mut map = self.value.clone();
```

Comment:
You're creating an unnecessary clone when the key is not present.  This version is OK, but could be optimized to not clone `self.value` if the key can't be removed.  You can do that here or leave it for a later PR.